docs/internal/architecture.md

```
@@ -0,0 +1,108 @@
+|-------|---------|--------|
+| `PASS` | `rundown pass` or command exit 0 | Evaluate PASS transition |
+| `FAIL` | `rundown fail` or command exit non-0 | Evaluate FAIL transition |
+| `GOTO` | `rundown goto N` or GOTO action | Jump to step N |
+| `RETRY` | FAIL + RETRY action | Increment retryCount, stay in state |
+
+These input events are distinct from the action output recorded as `lastAction.type` after a transition resolves. The full `LastAction` union has nine variants: `START`, `CONTINUE`, `DEFER`, `GOTO`, `COMPLETE`, `STOP`, `RETRY`, `NEXT`, `BREAK` (see `packages/core/src/runbook/types.ts`).
+
+### Transitions
+
+Default transitions when none specified:
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)